src/main/java/org/traccar/protocol/NavtelecomProtocolDecoder.java

```diff
@@ -305,9 +385,42 @@ protected Object decode(
                                 case 83:
                                     position.set("fuelTemp" + (j + 2 - 78), (int) buf.readByte());
                                     break;
-                                default:
-                                    buf.skipBytes(getItemLength(j + 1));
+                                case 163:
+                                case 164:
+                                case 165:
+                                case 166:
+                                    value = buf.readShortLE();
+                                    position.set(
+                                            Position.PREFIX_TEMP + (j + 2 + 8 - 163),
+                                            (value != (short) 0x8000) ? value * 0.05 : null);
                                     break;
+                                case 167:
+                                case 168:
+                                case 169:
+                                case 170:
+                                    value = buf.readUnsignedByte();
+                                    position.set("humidity" + (j + 2 - 167), (value != 0xFF) ? value * 0.5 : null);
+                                    break;
+                                case 206:
+                                    position.set("diagnostic", buf.readUnsignedIntLE());
+                                    break;
+                                default:
+                                    if ((207 <= j) && (j <= 222)) {
+                                        position.set("user1Byte" + (j + 2 - 207), buf.readUnsignedByte());
+                                        break;
+                                    } else if ((223 <= j) && (j <= 237)) {
+                                        position.set("user2Byte" + (j + 2 - 223), buf.readUnsignedShortLE());
+                                        break;
+                                    } else if ((238 <= j) && (j <= 252)) {
+                                        position.set("user4Byte" + (j + 2 - 238), buf.readUnsignedIntLE());
+                                        break;
+                                    } else if ((253 <= j) && (j <= 255)) {
+                                        position.set("user8Byte" + (j + 2 - 253), buf.readLongLE());
+                                        break;
```

Comment:
It looks good, but please remove the extra brackets in if statements.

Reply:
Ok, I'll fix